PurchasesCoreSwift/Networking/HTTPClient.swift

```diff
@@ -0,0 +1,306 @@
+    @objc public func performRequest(_ httpMethod: String,
+                                     performSerially: Bool = false,
+                                     path: String,
+                                     requestBody: [String: Any]?,
+                                     headers authHeaders: [String: String],
+                                     completionHandler: ((Int, [AnyHashable: Any]?, Error?) -> Void)?) {
+        performRequest(httpMethod,
+                       performSerially: performSerially,
+                       path: path,
+                       requestBody: requestBody,
+                       authHeaders: authHeaders,
+                       retried: false,
+                       completionHandler: completionHandler)
+    }
+
+    @objc public func clearCaches() {
+        operationDispatcher.dispatchOnHTTPSerialQueue { [self] in
+            self.eTagManager.clearCaches()
+        }
+    }
+
+}
+
+private extension HTTPClient {
+
+    var defaultHeaders: [String: String] {
+        let observerMode = systemInfo.finishTransactions ? "false" : "true"
+        var headers: [String: String] = [
+            "content-type": "application/json",
+            "X-Version": SystemInfo.frameworkVersion,
+            "X-Platform": SystemInfo.platformHeader,
+            "X-Platform-Version": SystemInfo.systemVersion,
+            "X-Platform-Flavor": systemInfo.platformFlavor,
+            "X-Client-Version": SystemInfo.appVersion,
+            "X-Client-Build-Version": SystemInfo.buildVersion,
+            "X-Observer-Mode-Enabled": observerMode
+        ]
+
+        if let platformFlavorVersion = self.systemInfo.platformFlavorVersion {
+            headers["X-Platform-Flavor-Version"] = platformFlavorVersion
+        }
+
+        if let idfv = SystemInfo.identifierForVendor {
+            headers["X-Apple-Device-Identifier"] = idfv
+        }
+        return headers
+    }
+
+    func performRequest(_ httpMethod: String,
+                        performSerially: Bool = false,
+                        path: String,
+                        requestBody maybeRequestBody: [String: Any]?,
+                        authHeaders: [String: String],
+                        retried: Bool = false,
+                        completionHandler maybeCompletionHandler: ((Int, [AnyHashable: Any]?, Error?) -> Void)?) {
+        operationDispatcher.dispatchOnHTTPSerialQueue { [self] in
+            do {
+                try self.assertIsValidRequest(httpMethod: httpMethod, requestBody: maybeRequestBody)
+            } catch let error {
+                if let completionHandler = maybeCompletionHandler {
+                    self.operationDispatcher.dispatchOnMainThread {
+                        completionHandler(-1, nil, error)
+                    }
+                    return
+                }
+            }
+
+            let requestHeaders = self.defaultHeaders.merging(authHeaders, uniquingKeysWith: { (_, last) in last })
```

Comment:
i was thinking during my PurchaserInfo migration... not necessarily in the scope of this PR, but could be nice to make a global extension on dictionary that defaults the `uniquingKeysWith` parameter to something.

i just called it a number of times the exact same way...

Reply:
Would you mind filling out an issue for this suggestion? I'd love for these types of things to be listed somewhere 😄